workflow/installertest.groovy

```diff
@@ -0,0 +1,133 @@
+// Replace colons in image with hyphens and append text
+String convertImageNameToString(String imageName, String append="") {
+    return (imageName+append).replaceAll(':','-')
+}
+
+/**
+ * Extracts the components from an {@code 'artifact://full/path/to/job/buildNr#artifact.ext'} type url.
+ * @param url the url
+ */
+@NonCPS
+def getComponentsFromArtifactUrl(String url) {
+    def pattern = /^artifact:\/\/([\/\w-_ ]+)\/(\d+)\/{0,1}#([\/\w\.]+)$/
```

Comment:
:bug: Just noticed a problem here - the item pattern doesn't include all valid characters for a job/folder, most notably `.`.